src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Block.vb

```diff
@@ -42,21 +42,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
 
             If Instrument Then
-                Dim prologue As BoundStatement = _instrumenter.CreateBlockPrologue(node)
+                Dim builder = ArrayBuilder(Of BoundStatement).GetInstance()
 
-                If prologue IsNot Nothing Then
-                    Dim builder = ArrayBuilder(Of BoundStatement).GetInstance()
-                    builder.Add(prologue)
-
-                    For Each s In node.Statements
-                        Dim rewrittenStatement = TryCast(Visit(s), BoundStatement)
-                        If rewrittenStatement IsNot Nothing Then
-                            builder.Add(rewrittenStatement)
-                        End If
-                    Next
+                For Each s In node.Statements
+                    Dim rewrittenStatement = TryCast(Visit(s), BoundStatement)
+                    If rewrittenStatement IsNot Nothing Then
+                        builder.Add(rewrittenStatement)
+                    End If
+                Next
 
-                    Return New BoundBlock(node.Syntax, node.StatementListSyntax, node.Locals, builder.ToImmutableAndFree())
+                Dim synthesizedLocal As LocalSymbol = Nothing
+                Dim prologue As BoundStatement = _instrumenter.CreateBlockPrologue(node, synthesizedLocal)
+                If prologue IsNot Nothing Then
+                    builder.Insert(0, prologue)
                 End If
```

Comment:
No, because the number of instrumentation points needs to be known before creating the prologue, and the number of instrumentation points is computed while instrumenting the statements in the method.